app/utils/decorator_utils.py

```diff
@@ -0,0 +1,41 @@
+"""
+This module is used to define decorators for the app
+"""
+from app.database.models.user import UserModel
+
+
+def email_verification_required(user_function):
+    """
+    This function is used to validate the
+    input function i.e. user_function
+    It will check if the user given as a
+    parameter to user_function
+    exists and have its email verified
+    """
+
+    def check_verification(*args, **kwargs):
+        """
+        Function to validate the input function ie user_function
+        - It will return error 404 if user doesn't exist
+        - It will return error 400 if user hasn't verified email
+
+        Parameters:
+        Function to be validated can have any type of argument
+        - list
+        - dict
+        """
+
+        if kwargs:
+            user = UserModel.find_by_id(kwargs['user_id'])
+        else:
+            user = UserModel.find_by_id(args[0])
```

Comment:
if we use `user_id` always instead of `assigner_user_id` and `revoker_user_id`, you won't need to assume that the first argument is always the `user_id`.
If you change to `user_id`, you avoid this hack

Reply:
@ramitsawhney27 does this `args[0]` make sense here since every time user ID appears on a  function it is name as `user_id` ? I'm not sure, maybe it does, I just want another opinion on this.
cc @LordGameleo

Reply:
We can name it explicitly before this statement. Something like
``
user_id = args[0]
user = UserModel.find_by_id(user _id)
`` 
Or we can just take input as 
``def check_verification(user_id, *args, **kwargs)``